### ISSUE TITLE
Segmentation fault on loss.grad_fn()

### ISSUE THREAD

**Issue by pyhh** [pytorch/pytorch#6278]
In the following code I get a segmentation fault when I evaluate `loss.grad_fn()`. Any ideas?

System:
- PyTorch or Caffe2: Pytorch
- OS: Ubuntu 16.04 LTS
- PyTorch version: 0.3.1.post2
- How you installed PyTorch (conda, pip, source): conda
- Python version: 3.6.4
- CUDA/cuDNN version: 9.0
- GPU models and configuration: Geforce GTX 970
- GCC version (if compiling from source): 5.4.0

Script:
```
import torch
from torch.autograd import Variable
import torch.nn as nn
import torch.nn.functional as F

class Net(nn.Module):

    def __init__(self):
        super(Net, self).__init__()
        # 1 input image channel, 6 output channels, 5x5 square convolution
        # kernel
        self.conv1 = nn.Conv2d(1, 6, 5)
        self.conv2 = nn.Conv2d(6, 16, 5)
        # an affine operation: y = Wx + b
        self.fc1 = nn.Linear(16 * 5 * 5, 120)
        self.fc2 = nn.Linear(120, 84)
        self.fc3 = nn.Linear(84, 10)

    def forward(self, x):
        # Max pooling over a (2, 2) window
        x = F.max_pool2d(F.relu(self.conv1(x)), (2, 2))
        # If the size is a square you can only specify a single number
        x = F.max_pool2d(F.relu(self.conv2(x)), 2)
        x = x.view(-1, self.num_flat_features(x))
        x = F.relu(self.fc1(x))
        x = F.relu(self.fc2(x))
        x = self.fc3(x)
        return x

    def num_flat_features(self, x):
        size = x.size()[1:]  # all dimensions except the batch dimension
        num_features = 1
        for s in size:
            num_features *= s
        return num_features

net = Net()
print(net)

params = list(net.parameters())
print(len(params))
print(params[0].size())  # conv1's .weight

input = Variable(torch.randn(1, 1, 32, 32), requires_grad=True)
out = net(input)
print(out)

net.zero_grad()
out.backward(torch.randn(1, 10))

output = net(input)
target = Variable(torch.arange(1, 11))  # a dummy target, for example
target = target.view(1, -1)  # make it the same shape as output
criterion = nn.MSELoss()

loss = criterion(output, target)
print(loss)

loss.grad_fn()
```

Trace:
```
Thread 1 "python" received signal SIGSEGV, Segmentation fault.
torch::autograd::generated::MseLossBackward::apply (this=0x555556819f48, grads=...)
    at torch/csrc/autograd/generated/Functions.cpp:2316
2316	torch/csrc/autograd/generated/Functions.cpp: No such file or directory.
(gdb) where
#0  torch::autograd::generated::MseLossBackward::apply (this=0x555556819f48, grads=...)
    at torch/csrc/autograd/generated/Functions.cpp:2316
#1  0x00007fffed020c0e in torch::autograd::Function::operator() (inputs=..., this=0x555556819f48)
    at /opt/conda/conda-bld/pytorch_1518244421288/work/torch/csrc/autograd/function.h:89
#2  torch::autograd::(anonymous namespace)::THPCppFunction_call (self=<optimised out>, 
    args=<optimised out>, kwargs=<optimised out>) at torch/csrc/autograd/python_cpp_function.cpp:45
#3  0x00005555556631bb in _PyObject_FastCallDict ()
#4  0x00005555556f0d3e in call_function ()
#5  0x000055555571519a in _PyEval_EvalFrameDefault ()
#6  0x00005555556eb529 in PyEval_EvalCodeEx ()
#7  0x00005555556ec2cc in PyEval_EvalCode ()
#8  0x0000555555768af4 in run_mod ()
#9  0x0000555555768ef1 in PyRun_FileExFlags ()
#10 0x00005555557690f4 in PyRun_SimpleFileExFlags ()
#11 0x000055555576cc28 in Py_Main ()
#12 0x000055555563471e in main ()
```
If I change the loss function to `loss = ((output - target) ** 2.0).sum()`, then I get the following:
```
Thread 1 "python" received signal SIGSEGV, Segmentation fault.
0x00007fffeda11285 in torch::autograd::generated::SumBackward0::apply(std::vector<torch::autograd::Variable, std::allocator<torch::autograd::Variable> > const&) ()
   from /home/user/anaconda3/lib/python3.6/site-packages/torch/_C.cpython-36m-x86_64-linux-gnu.so
(gdb) bt
#0  0x00007fffeda11285 in torch::autograd::generated::SumBackward0::apply(std::vector<torch::autograd::Variable, std::allocator<torch::autograd::Variable> > const&) ()
   from /home/user/anaconda3/lib/python3.6/site-packages/torch/_C.cpython-36m-x86_64-linux-gnu.so
#1  0x00007fffed970aa8 in torch::autograd::(anonymous namespace)::THPCppFunction_call(_object*, _object*, _object*) ()
   from /home/user/anaconda3/lib/python3.6/site-packages/torch/_C.cpython-36m-x86_64-linux-gnu.so
#2  0x00005555556631bb in _PyObject_FastCallDict ()
#3  0x00005555556f0d3e in call_function ()
#4  0x000055555571519a in _PyEval_EvalFrameDefault ()
#5  0x00005555556eb529 in PyEval_EvalCodeEx ()
#6  0x00005555556ec2cc in PyEval_EvalCode ()
#7  0x0000555555768af4 in run_mod ()
#8  0x0000555555768ef1 in PyRun_FileExFlags ()
#9  0x00005555557690f4 in PyRun_SimpleFileExFlags ()
#10 0x000055555576cc28 in Py_Main ()
#11 0x000055555563471e in main ()
```

**Comment by zou3519** [comment]
Why are you using `loss.grad_fn()`?

**Comment by pyhh** [comment]
I'm just being stupid by thinking it was callable, issue closed.
